staging_service/app.py

```diff
@@ -12,10 +15,66 @@
 from .globus import assert_globusid_exists, is_globusid
 from .metadata import some_metadata, dir_info, add_upa, similar
 from .utils import Path, run_command, AclManager
+from .import_specifications.file_parser import (
+    ErrorType,
+    Error,
+    FileTypeResolution,
+    parse_import_specifications,
+)
+from .import_specifications.individual_parsers import parse_csv, parse_tsv, parse_excel
+from .autodetect.Mappings import CSV, TSV, EXCEL
 
+
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
 routes = web.RouteTableDef()
 VERSION = "1.2.0"
 
+_APP_JSON = "application/json"
+
+_IMPSPEC_FILE_TO_PARSER = {
+    CSV: parse_csv,
+    TSV: parse_tsv,
+    EXCEL: parse_excel,
+}
+
+# Types for the lambda parameters are all strings
+# We don't add hints in to avoid repeating them over and over
+_ERROR_FORMATTERS = {
```

Comment:
Otherwise LGTM

Reply:
Not 100% sure, but I think app_factory requires the routes to be decorated in the same file, otherwise it won't know about them?
So there are 3
* app_factory 
* authorize_request (Maybe could pass this as a middleware in __main__.py (see https://docs.aiohttp.org/en/stable/web_advanced.html#middleware-factory) )
* inject_config_dependencies (Maybe it can be moved to __main__.py)

Reply:
I think I'm on the side of cohesion here. I'm guessing you're talking about `_file_type_resolver` and `_format_errors` and all the lambdas that go with that, to keep things decomposed?

It seems like you could do one of three things.
1. Leave as-is, keep the single-use code together. I don't think I favor this one that much.
2. Migrate the whole function (not just the non-route subfunctions) and its subfunctions into its own module and call out to it here with some wrapper for the route-decorated function. I'm generally in favor of this for servers - keep the functional stuff decomposed out, and deal with request/response stuff in the server module.
3. Move the error formatter code only. I think it could be more generally useful to other paths with a little adjustment (or maybe as it is). I can see a near future where we need to parse other staged files (say, for bulk execution), and this would be handy to just have around for importing.

Really, I'm kinda surprised `_file_type_resolver` isn't part of the `import_specifications` module already.

FWIW, the non-route functions could be in a separate module. Probably not `app_factory`, as it makes sense to keep here in `app.py`, but you could move it and have the routes as an input. Both `inject_config_dependencies` and `authorize_request` could move, though. And that `utils.Path` config injection smells weird to me anyway, though I've only glanced at it. But these are things for another PR.

Reply:
> Really, I'm kinda surprised _file_type_resolver isn't part of the import_specifications module already.

That function is really just glue code between the import spec module and the auto detect utils. With the current design, both of those modules are totally unaware of each other, and the resolver does all the dirty work of getting them to talk

Reply:
Ok, based on the Slack discussion with @briehl I moved the error formatting stuff into a new module, but left the file resolver alone (e.g. #3).